cymetric/metrics.py

```diff
@@ -504,4 +504,52 @@ def inventory_quantity_per_gwe(expinv,power):
     inv.Quantity = inv.Quantity/inv.Value
     inv=inv.drop(['Value'],axis=1)
     return inv
+  
+# Quantity per GigaWattElectric in TransactionQuantity [kg/GWe]
+_tranactsdeps = ['TransactionQuantity','TimeSeriesPower']
+
+_tranactsschema = [
+    ('SimId', ts.UUID),
+    ('TransactionId', ts.INT),
+    ('ResourceId', ts.INT),
+    ('ObjId', ts.INT),
+    ('Time', ts.INT),
+    ('SenderId', ts.INT),
+    ('ReceiverId', ts.INT),
+    ('Commodity', ts.STRING),
+    ('Units', ts.STRING),
+    ('Quantity', ts.DOUBLE)
+    ]
+
+
+@metric(name='TransactionQuantityPerGWe', depends=_tranactsdeps, schema=_tranactsschema)
+def transaction_quantity_per_gwe(tranacts,power):
+    """Transaction Quantity per GWe metric returns the transaction quantity table with quantity
+    in units of kg/GWe, calculated by dividing the original quantity by the electricity generated
+    at the corresponding simulation and the specific time in TimeSeriesPower metric.
+    """
+    power = pd.DataFrame(data={'SimId': power.SimId,
+			      'AgentId': power.AgentId,
+                              'Time': power.Time,
+                              'Value': power.Value},
+			columns=['SimId','AgentID','Time', 'Value'])
+    power_index = ['SimId','Time']
+    power = power.groupby(power_index).sum()
+    df1 = power.reset_index()
+    tranacts = pd.DataFrame(data={'SimId': tranacts.SimId,
+			     'TransactionId': tranacts.TransactionId,
+			     'ResourceId': tranacts.ResourceId,
+                             'ObjId': tranacts.ObjId,
+			     'Time': tranacts.TimeCreated,
+                             'SenderId': tranacts.SenderId, 
+			     'ReceiverId': tranacts.ReceiverId,
+		             'Commodity': tranacts.Commodity,
+	 		     'Units': tranacts.Units,
+                             'Quantity': tranacts.Quantity},
+	             columns=['SimId','TransactionId','ResourceId','ObjId','Time','SenderId','ReceiverId','Commodity','Units','Quantity'])
+    tranacts['Units'] = "kg/GWe"    
+    tranacts=pd.merge(tranacts,df1, on=['SimId','Time'],how='left')
+    tranacts.Quantity = tranacts.Quantity/tranacts.Value
+    tranacts=tranacts.drop(['Value'],axis=1)
```

Comment:
you actually don't need to drop this, you can just return only some column of the pdf using something like:
`return tranacts[ ['SimId', 'TransactionId' ,..., 'Commodity', 'Units', 'Quantity' ] ]`